cmd/collector/app/metrics.go

```diff
@@ -41,10 +45,11 @@ type SpanProcessorMetrics struct {
 	// QueueLength measures the size of the internal span queue
 	QueueLength metrics.Gauge
 	// SavedOkBySvc contains span and trace counts by service
-	SavedOkBySvc  metricsBySvc  // spans actually saved
-	SavedErrBySvc metricsBySvc  // spans failed to save
-	serviceNames  metrics.Gauge // total number of unique service name metrics reported by this collector
-	spanCounts    map[string]CountsBySpanType
+	SavedOkBySvc      metricsBySvc  // spans actually saved
+	SavedErrBySvc     metricsBySvc  // spans failed to save
+	serviceNames      metrics.Gauge // total number of unique service name metrics reported by this collector
+	spanCounts        map[string]CountsBySpanType
+	countsByEndpoints map[string]metrics.Counter // count of spans processed from different endpoints (Http, TChannel, gRPC)
```

Comment:
I am not seeing TChannel being passed anywhere.

Reply:
I'm not sure what's the best way to detect if request is from tchannel. Is there something we passed along span/trace/context/etc that has this kind of information?